modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/DataTypesTest.java

```diff
@@ -98,10 +103,13 @@ private void testUuid(boolean indexed) {
                 .returns(2, uuid2.toString(), uuid2)
                 .check();
 
-            assertQuery("SELECT t1.* from t t1, (select * from t) t2 where t1.uid = t2.uid")
-                .returns(1, "fd10556e-fc27-4a99-b5e4-89b8344cb3ce", uuid1)
-                .returns(2, uuid2.toString(), uuid2)
-                .check();
+            // TODO: IGNITE-16693 - Skip merge join. Incorrect processing of null values.
```

Comment:
Full link to ticket, let's add a hint to disable merge join.

Reply:
Fixed